### PR TITLE
Add more documentation to `bbi_init`: make `.nonmem_dir` argument more clear

### PR DESCRIPTION
**Pull request by barrettk**
- It became apparent that we needed more clear documentation surrounding the NONMEM installation directory (`.nonmem_dir` argument), as users were incorrectly passing the directory containing the executable file, rather than the parent directory.

 - Added more context to the parameter itself, along with some additional details and an example.

closes https://github.com/metrumresearchgroup/bbr/issues/607

### CODE REVIEW

**Comment by barrettk** [comment]
@kyleam @seth127 I imagine we'll have a little back and forth regarding the documentation updates. Requesting you both to review in the event you both want to give your two cents for added clarity regarding this issue. In all honesty the confusion seems well founded, and I would probably have assumed the same based on the documentation if I hadnt worked with `bbr` as much. Point being I want to make sure we are as clear as possible, as I would intuitively thing to pass the directory containing the executable, rather than the parent directory. I think the example and details help, but let me know if it's too wordy or not concise enough.

File: R/bbr.R [hunk 1]
```diff
@@ -321,9 +321,22 @@ bbi_help <- function(.cmd_args=NULL) {
 #'   model. `submit_model()` and `submit_models()` also support passing a
 #'   configuration file via the `.config_path` argument.
 #'
+#'  **Notes on `.nonmem_dir`**
+#'
+#'   `bbi` tries to detect subdirectories for different NONMEM versions, along
+#'   with a license file and executable within the subdirectory. `.nonmem_dir`
+#'   should therefore point to the *parent* directory containing the executable.
```
Comment (kyleam) [inline R/bbr.R:328]: Hmm, I'm not sure what "the executable" is here (I know you expand on it a bit in the next paragraph, but I'm explaining how I'd be confused by it as a reader).  The parent directory that should be passed as `.nonmem_dir` will typically just contain (assuming contain means first-level contents) one or more directories for the NONMEM installation.

Even _within_ a NONMEM installation's `run/` directory, I think saying "the executable" would be confusing because there are several executables.  (The one bbi cares about is `nmfe*`, but I don't think we should get into that level of detail here.)

File: R/bbr.R [hunk 2]
```diff
@@ -321,9 +321,22 @@ bbi_help <- function(.cmd_args=NULL) {
...
+#'
+#'   If your NONMEM executable exists in the the location `/opt/NONMEM/nm75/run`,
```
Comment (kyleam) [inline R/bbr.R:330]: typo: "the the"

Comment (kyleam) [inline R/bbr.R:330]: "exists in the location `/opt/NONMEM/nm75/run`"  here must mean under the `run/` directory, but that could be interpreted as though run itself is the executable file of interest.

I wonder if this is the kind of thing were an example layout would help a lot.  For example

```
**Notes on `.nonmem_dir`**

`.nonmem_dir` should point to the parent directory that contains one
or more NONMEM installations.  It should not point to the installation
directory itself.  Here is an example:

    /opt/NONMEM/
    |-- nm74
    |   |-- license
    |   |-- run
    |   |-- source
    |   | ...
    |   `-- util
    `-- nm75
        |-- license
        |-- run
        |-- source
        | ...
        `-- util

`/opt/NONMEM/` contains two NONMEM installations, one for NONMEM 7.5
and one for NONMEM 7.5.  In this case, you should pass "/opt/NONMEM"
as `.nonmem_dir` and either "nm74" or "nm75" as `.nonmem_version`.
```

File: R/bbr.R [hunk 3]
```diff
@@ -338,6 +351,18 @@ bbi_help <- function(.cmd_args=NULL) {
 #'
 #' @importFrom yaml read_yaml write_yaml
 #' @importFrom fs dir_exists
+#'
+#' @examples
+#' \dontrun{
+#'
+#' # NONMEM executable found at `/opt/NONMEM/nm75/run`:
```
Comment (kyleam) [inline R/bbr.R:358]: Same comment about it not being confusing what executable this is referring to.  Perhaps "# Top-level of NONMEM 7.5 installation is at `/opt/NONMEM/nm75`:"?

File: R/bbr.R [hunk 4]
```diff
@@ -338,6 +351,18 @@ bbi_help <- function(.cmd_args=NULL) {
...
+#'
+#' bbi_init(
+#'    .dir = "inst/model/nonmem/basic",
```
Comment (kyleam) [inline R/bbr.R:361]: From the bbr checkout, using this `inst/` path is of course handy to test, but that's an unusual directory from the user perspective.  What about just using a `MODEL_DIR` stand-in?  That would align with what's used expo1.

**Review by kyleam — CHANGES_REQUESTED** [review]
Thanks for working on this.  I think this is a clear improvement and would help avoid the confusion.

I'm leaving several comments and suggestions, but the only thing that I consider a blocker is revising the parts that talk about "the executable".

> as users were incorrectly passing the directory containing the executable file, rather than the parent directory.

Hmm, I'm confused by this part of the commit message.  I assume this is talking about the nmfe executable under the installations `run/` subdirectory.  In that case, I don't think this accurately captures the problem.  I think most users that were passing the wrong directory were passing the top-level installation directory (say `/path/to/nm-install/`) not the directory containing the nmfe executable (`/path/to/nm-install/run/`).

File: R/bbr.R (hunk 1 above)
Comment (barrettk) [reply R/bbr.R:328]: I was trying to reword your comment [here](https://github.com/metrumresearchgroup/bbr/issues/596#issuecomment-1548797780):

> Underneath, bbi init looks through .nonmem_dir and tries to detect subdirectories for different NONMEM versions ([code](https://github.com/metrumresearchgroup/bbi/blob/f8a2c83869381a2d1221aeb302da880093079b91/cmd/init.go#L39-L76)). Under each of those subdirectories, it looks for things like expected subdirectories, a license file, and an NONMEM executable.

I think the example layout would definitely help alot, but was worried we wouldnt want to mention Metworx. Since you posted that example without mentioning Metworx ill definitely add that. I do think it adds clarity to mention the executable, or some file that we can point to and know it's 2 directories up from there. I'll try to rephrase and potentially remove that word altogether, but I thought your comment on that issue added some clarity

Comment (kyleam) [reply R/bbr.R]: > I was trying to reword your comment [...]

I see.  Yeah, the original comment is trying to distinguish a couple levels of directories (e.g., "Under of of those subdirectories ..."), and I think in the comment here it'd be unclear to me that it's talking about subdirectories within the NONMEM installation directory.

> an NONMEM executable.

Aside from my "an" typo, yeah, I was a bit sloppy there to say "NONMEM executable" and since I was describing in detail what bbi does, I should have specified `nmfe*`.  Again, in my eyes that a bit too much detail for here, but I'm okay with what you decide there.  The main thing I'm trying to avoid here is statements like "`.nonmem_dir` should therefore point to the *parent* directory containing the executable"; I have a hard time making that correct unless "containing" is interpreted recursively (which I think would be surprisingly).

File: R/bbr.R (hunk 2 above)
Comment (barrettk) [reply R/bbr.R]: I left the first section, but removed the second mention of the executable in favor of your example layout

File: R/bbr.R (hunk 4 above)
Comment (barrettk) [reply R/bbr.R]: Yeah I think that makes sense. Addressed.

File: R/bbr.R (hunk 3 above)
Comment (barrettk) [reply R/bbr.R]: addressed

File: R/bbr.R (hunk 1 above)
Comment (barrettk) [reply R/bbr.R]: I only have one mention of the executable in my most recent commit. Let me know if it be helpful to add `nmfe` before that mention, or if we should reword that sentence altogether. I kind of like mentioning those contents in the event external scientists are more familiar with specific NONMEM installation files, but definitely want it to be as obvious as possible

File: R/bbr.R [hunk 5]
```diff
@@ -321,9 +321,41 @@ bbi_help <- function(.cmd_args=NULL) {
 #'   model. `submit_model()` and `submit_models()` also support passing a
 #'   configuration file via the `.config_path` argument.
 #'
+#'  **Notes on `.nonmem_dir`**
+#'
+#'   `bbi` tries to detect subdirectories for different NONMEM versions, along
+#'   with a license file and executable within the subdirectory.
```
Comment (kyleam) [inline R/bbr.R:327]: My preference would be to drop this entirely.  What `bbi` looks for to decide a directory is a NONMEM installation is an implementation detail, and I find it a bit confusing as a lead off paragraph.  For example

> `bbi` tries to detect subdirectories for different NONMEM versions

Where does it look for these?  The reader is supposed to infer `.nonmem_dir`, at this point they haven't even been given the details below about htat being the parent directory.

> along with a license file and executable within the subdirectory

it's not clear to me which directory "the subdirectory" is referring to.

I think the layout below makes it clear what `.nonmem_dir` should be, and this first paragraph is more likely to cause confusion.

File: R/bbr.R [hunk 6]
```diff
@@ -321,9 +321,41 @@ bbi_help <- function(.cmd_args=NULL) {
...
+#'   `/opt/NONMEM/` contains two NONMEM installations, one for NONMEM 7.5
+#'   and one for NONMEM 7.4.  In this case, you should pass `"/opt/NONMEM"`
+#'   as `.nonmem_dir` and either `"nm74"` or `"nm75"` as `.nonmem_version`.
+#'
+#'
 #' @param .dir Path to directory to run `init` in (and put the resulting
 #'   `bbi.yaml` file)
-#' @param .nonmem_dir Path to directory with the NONMEM installation.
+#' @param .nonmem_dir Path to directory with the NONMEM installation. This
+#'   should be the **parent** directory that contains that installation. See
```
Comment (kyleam) [inline R/bbr.R:357]: [ This was suppose to be a part of my first review, but I don't see it; I must have forgot to post it.  Sorry. ]

The second sentence would point people in the right direction, but I find the combination of the first and second sentences a bit confusing.  To me it almost reads as the though the second sentence corrects what the first just said.  (That all depends on how the reader interprets "directory with", which I suppose is probably the core of the original issue.)

I'd be in favor of reducing the first two sentences down to one.  Perhaps "Path to parent directory containing one or more NONMEM installations. See [...]".

**Review by kyleam — APPROVED** [review]
Thanks for the updates. 

I think this PR makes things this argument much clearer.  

Approving, but please see my second comment because that was supposed to come with my _first_ review, but I somehow left it out.

**Comment by seth127** [comment]
Thanks for all the work on this, both of you. I think this is much better at this point. I especially think [showing the potential directory layout](https://github.com/metrumresearchgroup/bbr/blob/f0aecbd53eb33d2e46ebafb5c7512db3c9eec6e1/R/bbr.R#L324-L351) will be very helpful. It would be for me, at least.

**Review by kyleam — APPROVED** [review]
